go-controller/pkg/clustermanager/networkconnect/controller.go

```diff
@@ -0,0 +1,544 @@
+package networkconnect
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	nadclientset "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned"
+	nadlisters "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/listers/k8s.cni.cncf.io/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/id"
+	controllerutil "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/controller"
+	networkconnectv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/clusternetworkconnect/v1"
+	networkconnectclientset "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/clusternetworkconnect/v1/apis/clientset/versioned"
+	networkconnectlisters "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/clusternetworkconnect/v1/apis/listers/clusternetworkconnect/v1"
+	apitypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/types"
+	userdefinednetworkv1 "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/userdefinednetwork/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/networkmanager"
+	ovntypes "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+var (
+	cudnGVK = userdefinednetworkv1.SchemeGroupVersion.WithKind("ClusterUserDefinedNetwork")
+	udnGVK  = userdefinednetworkv1.SchemeGroupVersion.WithKind("UserDefinedNetwork")
+)
+
+// clusterNetworkConnectState is the cache that keeps the state of a single
+// cluster network connect in the cluster with name being unique
+type clusterNetworkConnectState struct {
+	// name of the cluster network connect (unique across cluster)
+	name string
+	// allocator for this CNC's subnet allocation
+	allocator HybridConnectSubnetAllocator
+	// map of NADs currently selected by this CNC's network selectors
+	// {value: NAD namespace/name key}
+	selectedNADs sets.Set[string]
```

Comment:
I am wondering if we can only track `selectedNetworks`, and then on ns/nad change get the network name (can be always defined from the nad info?) and see if it started/stopped matching?
Then we would skip same-network updates coming from different NADs?

Reply:
ah I see, can you add a comment about that to the field then?